src/vmm/src/devices/virtio/queue.rs

```diff
@@ -839,20 +861,10 @@ mod verification {
         /// fixed to a known valid one
         pub fn bounded_queue() -> Self {
             let mem = setup_kani_guest_memory();
-            let queue = less_arbitrary_queue();
-
-            assert!(queue.is_layout_valid(&mem));
-
-            ProofContext(queue, mem)
-        }
-
-        /// Creates a [`ProofContext`] where the queue layout is fixed to a valid one and where
-        /// guest memory is initialized to all zeros.
-        pub fn bounded() -> Self {
-            let mem = setup_zeroed_guest_memory();
-            let queue = less_arbitrary_queue();
+            let mut queue = less_arbitrary_queue();
+            queue.initialize(&mem).unwrap();
```

Comment:
I think you'll have to inline `initialize` here if you want to keep the assertion. Otherwise kani is gonna keep hitting it (and we don't want to `kani::assume` that we don't hit it outside of the actual proof for `pop` because technically all those other functions can be called with the queue in a "guest messed it up" state).

Reply:
What about marking assert as `cfg(not(kani))`?

Reply:
not a huge fan, but if you strongly prefer it, sure

Reply:
done